core/src/main/java/jenkins/model/ModelObjectWithPageMenu.java

```diff
@@ -0,0 +1,44 @@
+package jenkins.model;
+
+import hudson.Functions;
+import hudson.model.ModelObject;
+import jenkins.model.ModelObjectWithContextMenu.ContextMenu;
+import org.kohsuke.stapler.StaplerRequest;
+import org.kohsuke.stapler.StaplerResponse;
+
+/**
+ * {@link ModelObject} that has a page menu in the breadcrumb bar or menu in its page link
+ *
+ * <p>
+ * The page context menu is accessible to the user via the breadcrumb bar or via a link
+ * to the object's page. The page menu combines the page's contextMenu and childrenContextMenu,
+ * making it easier for users to navigate to all the ancestor objects this model object owns,
+ * and its children.
+ *
+ * @author Jan Faracik, Lewis Birks
+ * @see ModelObjectWithContextMenu
+ * @since TODO - Provide version
+ */
+public interface ModelObjectWithPageMenu extends ModelObject {
```

Comment:
This looks like a UI implementation detail better not reflected in the type hierarchy, why is this entire thing not just in `Functions`?